src/vehicle_gui_base.h

```diff
@@ -14,32 +14,54 @@
 #include "date_type.h"
 #include "economy_type.h"
 #include "sortlist_type.h"
+#include "vehicle_base.h"
 #include "vehiclelist.h"
 #include "window_gui.h"
 #include "widgets/dropdown_type.h"
 
 #include <iterator>
+#include <numeric>
 
 typedef GUIList<const Vehicle*> GUIVehicleList;
 
 struct GUIVehicleGroup {
 	VehicleList::const_iterator vehicles_begin;    ///< Pointer to beginning element of this vehicle group.
 	VehicleList::const_iterator vehicles_end;      ///< Pointer to past-the-end element of this vehicle group.
-	Money display_profit_this_year;                ///< Total profit for the vehicle group this year.
-	Money display_profit_last_year;                ///< Total profit for the vehicle group laste year.
-	Date age;                                      ///< Age in days of oldest vehicle in the group.
 
-	GUIVehicleGroup(VehicleList::const_iterator vehicles_begin, VehicleList::const_iterator vehicles_end, Money display_profit_this_year, Money display_profit_last_year, Date age)
-		: vehicles_begin(vehicles_begin), vehicles_end(vehicles_end), display_profit_this_year(display_profit_this_year), display_profit_last_year(display_profit_last_year), age(age) {}
+	GUIVehicleGroup(VehicleList::const_iterator vehicles_begin, VehicleList::const_iterator vehicles_end)
+		: vehicles_begin(vehicles_begin), vehicles_end(vehicles_end) {}
 
 	std::ptrdiff_t NumVehicles() const
 	{
-		return std::distance(vehicles_begin, vehicles_end);
+		return std::distance(this->vehicles_begin, this->vehicles_end);
 	}
+
 	const Vehicle *GetSingleVehicle() const
 	{
-		assert(NumVehicles() == 1);
-		return vehicles_begin[0];
+		assert(this->NumVehicles() == 1);
+		return this->vehicles_begin[0];
+	}
+
+	Money GetDisplayProfitThisYear() const
+	{
+		return std::accumulate(this->vehicles_begin, this->vehicles_end, (Money)0, [](Money acc, const Vehicle *v) {
+			return acc + v->GetDisplayProfitThisYear();
```

Comment:
Was just reading what `std::accumulate` does, and they suggest to mark `acc` with `std::move` in the binary-op, but you haven't here. (well, in C++20 this is done for you, it seems, but we are not there yet :P ). Is there a drawback of not doing this?

Reply:
I'm pretty sure it's unnecessary in this case, unless my test code is too simple... https://godbolt.org/z/sxo97o

Reply:
Empirical evidence :P

I wouldn't have a clue what is true or not, just noticed the manual suggests using it, and it being default for C++20. I will leave it for someone who knows about C++ to judge this :D

Reply:
In any case, the first parameter of the lambda function is `Money acc` which is passed by value already, so the copy is already done in `std::accumulate` (before C++20) before control reaches the lambda function.  Writing `std::move(acc) + v->GetDisplayProfitThisYear()` won't get rid of __that__ move, which is the thing that C++20 changes.

Reply:
To quote my dearest C++ expert: "Move semantics are only relevant if copying is expensive. For a type that is not larger than a pointer and has no constructor/destructor, it has no practical meaning."